modin/data_management/data_manager.py

```diff
@@ -0,0 +1,2547 @@
+
+    def mode(self, **kwargs):
+        """Returns a new DataManager with modes calculated for each label along given axis.
+
+        Returns:
+            A new PandasDataManager with modes calculated.
+        """
+        axis = kwargs.get("axis", 0)
+        func = self._prepare_method(pandas.DataFrame.mode, **kwargs)
+        new_data = self.map_across_full_axis(axis, func)
+
+        counts = self.__constructor__(new_data, self.index,
+                                      self.columns).notnull().sum(axis=axis)
+        max_count = counts.max()
+
+        new_index = pandas.RangeIndex(max_count) if not axis else self.index
+        new_columns = self.columns if not axis else pandas.RangeIndex(
+            max_count)
+
+        # We have to reindex the DataFrame so that all of the partitions are
+        # matching in shape. The next steps ensure this happens.
+        final_labels = new_index if not axis else new_columns
+        # We build these intermediate objects to avoid depending directly on
+        # the underlying implementation.
+        final_data = self.__constructor__(
+            new_data, new_index, new_columns).map_across_full_axis(
+                axis, lambda df: df.reindex(axis=axis, labels=final_labels))
+        return self.__constructor__(final_data, new_index, new_columns,
+                                    self._dtype_cache)
+
+    def fillna(self, **kwargs):
+        """Returns a new DataManager with null values filled by given values or according to given method.
+
+        Returns:
+            A new PandasDataManager with null values filled.
+        """
+        axis = kwargs.get("axis", 0)
+        value = kwargs.get("value")
+
+        if isinstance(value, dict):
+            value = kwargs.pop("value")
+
+            if axis == 0:
+                index = self.columns
+            else:
+                index = self.index
+            value = {
+                idx: value[key]
+                for key in value for idx in index.get_indexer_for([key])
+            }
+
+            def fillna_dict_builder(df, func_dict={}):
+                return df.fillna(value=func_dict, **kwargs)
+
+            new_data = self.data.apply_func_to_select_indices(
+                axis, fillna_dict_builder, value, keep_remaining=True)
+            return self.__constructor__(new_data, self.index, self.columns)
+        else:
+            func = self._prepare_method(pandas.DataFrame.fillna, **kwargs)
+            new_data = self.map_across_full_axis(axis, func)
+            return self.__constructor__(new_data, self.index, self.columns)
+
+    def query(self, expr, **kwargs):
+        """Query columns of the DataManager with a boolean expression.
+
+        Args:
+            expr: Boolean expression to query the columns with.
+
+        Returns:
+            DataManager containing the rows where the boolean expression is satisfied.
+        """
+        columns = self.columns
+
+    
```

Comment:
Are the managers returned specifically PANDASDataManagers? Or just DataManagers?